docs/decisions/0013-library-reference-content-block.rst

```diff
@@ -0,0 +1,62 @@
+Referencing Content Blocks in Library V2
+--------------------------------------------------
+
+Status
+=======
+Pending
+
+Context
+=======
+Currently, the courses are being stored in the Modulestore and the libraries are stored in both Modulestore and the Blockstore. We want the library content to be stored in the Blockstore.
+To achieve this we need a referencing mechanism to reference the libraries to the courses. This referencing helps in replacing
+v1 libraries (current content libraries stored on Modulestore) with v2 libraries (Blockstore backed) which are:
+
+We are currently building out the functionality of libraries v2, and trying to migrate all usage of content libraries onto Blockstore.
+The benefits of using the Blockstore backed v2 libraries are:
+
+#. Blockstore is much more flexible than modulestore and can store anything that can be represented as a file - unlike modulestore, which only stores courses built of XBlocks. Also, the Blockstore is being moved into the edx-platform.
+#. simpler to maintain than modulestore
+#. increases re-usability of the content
+
+Terminology Used
+^^^^^^^^^^^^^^^^
+#. Blocks: Each reusable piece of content in a library. Example: problems, videos, HTML. Also known as Xblocks.
+#. Library: Collection of blocks in a course.
+
+
+Decisions
+=========
+We want to be able to reference single and multiple specific blocks from a library (v2, as well as v1) into a course (that are stored in modulestore)
+The workflow discussed for the interface for library content referencing would be:
+
+#. Author chooses a library.
+#. Author chooses a pool of blocks.
+#. If multiple blocks of the same type are selected, ask for randomization
+#. If no randomization, then ask for ordering
+
+Current Architecture/Implementation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Currently, the courses are stored in modulestore and the libraries can either be stored in modulestore or blockstore.
+Since the course lives in the modulestore and some of the libraries in blockstore, we use the [library_sourced_block](https://github.com/openedx/edx-platform/blob/master/common/lib/xmodule/xmodule/library_sourced_block.py) to make a copy of that blockstore based library and store it in modulestore itself as the child.
```

Comment:
Missing word: "make a copy of that blockstore based library **block** and store it"

It doesn't copy the whole library; it just copies a block: